### PR TITLE
fix(gateway): don't log IllegalArgumentException as errors

### LINKED ISSUE #9832: MessagePack cannot serialize BigInteger larger than 2^64-1
**Describe the bug**

An [error](https://console.cloud.google.com/errors/detail/CJSquanBzMSL-AE;service=zeebe;filter=%5B%5D;time=P7D?project=camunda-cloud-240911) was reported: **RuntimeException: Failed to convert JSON to MessagePack.**

Caused by: `java.lang.IllegalArgumentException: MessagePack cannot serialize BigInteger larger than 2^64-1`

This is just another in a long list of issues with the MessagePack conversion:
- #7187
- #7025
- #5565
- #1441 

**To Reproduce**

Just run the following: 

`zbctl create instance 123 --variables='{"mybigintistoolong":123456789012345678901234567890}'`
>Error: rpc error: code = Internal desc = Unexpected error occurred during the request processing: Failed to convert JSON to MessagePack

**Expected behavior**

User input should not lead to reported errors, but user command should be rejected with a helpful message.

**Log/Stacktrace**

<details><summary>Full Stacktrace</summary>
 <p>

```
 java.lang.RuntimeException: Failed to convert JSON to MessagePack
	at io.camunda.zeebe.protocol.impl.encoding.MsgPackConverter.convertToMsgPack(MsgPackConverter.java:80) ~[zeebe-protocol-impl-8.0.3.jar:8.0.3]
	at io.camunda.zeebe.protocol.impl.encoding.MsgPackConverter.convertToMsgPack(MsgPackConverter.java:71) ~[zeebe-protocol-impl-8.0.3.jar:8.0.3]
	at io.camunda.zeebe.gateway.RequestMapper.ensureJsonSet(RequestMapper.java:175) ~[zeebe-gateway-8.0.3.jar:8.0.3]
	at io.camunda.zeebe.gateway.RequestMapper.toCreateProcessInstanceRequest(RequestMapper.java:113) ~[zeebe-gateway-8.0.3.jar:8.0.3]
	at io.camunda.zeebe.gateway.EndpointManager.sendRequestWithRetryPartitions(EndpointManager.java:353) ~[zeebe-gateway-8.0.3.jar:8.0.3]
	at io.camunda.zeebe.gateway.EndpointManager.createProcessInstance(EndpointManager.java:177) ~[zeebe-gateway-8.0.3.jar:8.0.3]
	at io.camunda.zeebe.gateway.GatewayGrpcService.createProcessInstance(GatewayGrpcService.java:77) ~[zeebe-gateway-8.0.3.jar:8.0.3]
	at io.camunda.zeebe.gateway.protocol.GatewayGrpc$MethodHandlers.invoke(GatewayGrpc.java:1597) ~[zeebe-gateway-protocol-impl-8.0.3.jar:8.0.3]
	at io.grpc.stub.ServerCalls$UnaryServerCallHandler$UnaryServerCallListener.onHalfClose(ServerCalls.java:182) ~[grpc-stub-1.45.1.jar:1.45.1]
	at io.grpc.PartialForwardingServerCallListener.onHalfClose(PartialForwardingServerCallListener.java:35) ~[grpc-api-1.45.1.jar:1.45.1]
	at io.grpc.ForwardingServerCallListener.onHalfClose(ForwardingServerCallListener.java:23) ~[grpc-api-1.45.1.jar:1.45.1]
	at io.grpc.internal.ServerCallImpl$ServerStreamListenerImpl.halfClosed(ServerCallImpl.java:340) ~[grpc-core-1.45.1.jar:1.45.1]
	at io.grpc.internal.ServerImpl$JumpToApplicationThreadServerStreamListener$1HalfClosed.runInContext(ServerImpl.java:866) ~[grpc-core-1.45.1.jar:1.45.1]
	at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37) ~[grpc-core-1.45.1.jar:1.45.1]
	at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:133) ~[grpc-core-1.45.1.jar:1.45.1]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) ~[?:?]
	at java.lang.Thread.run(Unknown Source) ~[?:?]
Caused by: java.lang.IllegalArgumentException: MessagePack cannot serialize BigInteger larger than 2^64-1
	at org.msgpack.core.MessagePacker.packBigInteger(MessagePacker.java:615) ~[msgpack-core-0.9.1.jar:0.9.1]
	at org.msgpack.jackson.dataformat.MessagePackGenerator.pack(MessagePackGenerator.java:230) ~[jackson-dataformat-msgpack-0.9.1.jar:0.9.1]
	at org.msgpack.jackson.dataformat.MessagePackGenerator.addValueToStackTop(MessagePackGenerator.java:591) ~[jackson-dataformat-msgpack-0.9.1.jar:0.9.1]
	at org.msgpack.jackson.dataformat.MessagePackGenerator.writeNumber(MessagePackGenerator.java:414) ~[jackson-dataformat-msgpack-0.9.1.jar:0.9.1]
	at com.fasterxml.jackson.core.JsonGenerator.copyCurrentEvent(JsonGenerator.java:2463) ~[jackson-core-2.13.2.jar:2.13.2]
	at com.fasterxml.jackson.core.JsonGenerator.copyCurrentStructure(JsonGenerator.java:2555) ~[jackson-core-2.13.2.jar:2.13.2]
	at io.camunda.zeebe.protocol.impl.encoding.MsgPackConverter.convert(MsgPackConverter.java:136) ~[zeebe-protocol-impl-8.0.3.jar:8.0.3]
	at io.camunda.zeebe.protocol.impl.encoding.MsgPackConverter.convertToMsgPack(MsgPackConverter.java:76) ~[zeebe-protocol-impl-8.0.3.jar:8.0.3]
	... 17 more 
```

</p>
</details>

**Environment:**
- Zeebe Version: 8.0.3, 8.0.4, etc.
- Both SaaS and SM

### PR DESCRIPTION
**Pull request by skayliu**
## Description

IllegalArgumentException get thrown when a user passes variables in their request which cannot be converted to message pack. Currently these are being logged as errors. Since this a fault by the user we should log these as debug.

User input should not lead to reported errors, but user command should be rejected with a helpful message.

## Related issues

closes #9832

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [x] The changes are backwards compatibility with previous versions
* [x] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [x] There are unit/integration tests that verify all acceptance criterias of the issue
* [x] New tests are written to ensure backwards compatibility with further versions
* [x] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

**Comment by skayliu** [comment]
Hi, @saig0, @korthout . Please take a moment to review this fix when you have time. Thank you.

**Comment by saig0** [comment]
@skayliu thank you for your contribution. :tada: We'll have a look at your PR in the next few days. :eyes:

File: protocol-impl/src/main/java/io/camunda/zeebe/protocol/impl/encoding/MsgPackConverter.java [hunk 1]
```diff
@@ -77,7 +77,11 @@ public static byte[] convertToMsgPack(final InputStream inputStream) {
 
       return outputStream.toByteArray();
     } catch (final Exception e) {
-      throw new RuntimeException("Failed to convert JSON to MessagePack", e);
+      if (e instanceof IllegalArgumentException) {
+        throw new IllegalArgumentException("Failed to convert JSON to MessagePack", e);
+      } else {
+        throw new RuntimeException("Failed to convert JSON to MessagePack", e);
+      }
```
Comment (remcowesterhoud) [inline protocol-impl/src/main/java/io/camunda/zeebe/protocol/impl/encoding/MsgPackConverter.java:84]: ❓ I'm a bit confused about the changes, why is this necessary? Is it only to change it to an `IllegalArgumentException` so we have a specific exception we can catch and not change the behavior of other `RuntimeExceptions`?

**Review by remcowesterhoud — CHANGES_REQUESTED** [review]
Hi @skayliu! Thanks for the changes, I have tested it locally and it all seems to be working fine. I have one question about the changes though, and a small comment.

🔧 Please also add a testcase to the `RequestMapperTest`

File: protocol-impl/src/main/java/io/camunda/zeebe/protocol/impl/encoding/MsgPackConverter.java (hunk 1 above)
Comment (skayliu) [reply protocol-impl/src/main/java/io/camunda/zeebe/protocol/impl/encoding/MsgPackConverter.java:84]: https://github.com/camunda/zeebe/issues/9832 got `IllegalArgumentException` exception, I'm not sure the others may get `RuntimeExceptions` exception  or not, so try specific `IllegalArgumentException ` exception to catch.

**Comment by skayliu** [comment]
> Please also add a testcase to the RequestMapperTest

@remcowesterhoud, Done, please check this out. Thank you.

**Review by remcowesterhoud — APPROVED** [review]
Thanks for the fix @skayliu 🙇 

bors merge

**Comment by skayliu** [comment]
Hi, @remcowesterhoud, It seems to need to rebase to main. Done, Please try again, Thank you.

**Review by remcowesterhoud — APPROVED** [review]
bors merge

**Comment by remcowesterhoud** [comment]
bors retry

**Pull request merged**